src/Features/CSharp/Portable/ConvertForToForEach/CSharpConvertForToForEachCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,151 @@
+
+        protected override SyntaxList<StatementSyntax> GetBodyStatements(ForStatementSyntax forStatement)
+            => forStatement.Statement is BlockSyntax block
+                ? block.Statements
+                : SyntaxFactory.SingletonList(forStatement.Statement);
+
+        protected override bool TryGetForStatementComponents(
+            ForStatementSyntax forStatement,
+            out SyntaxToken iterationVariable, out ExpressionSyntax initializer,
+            out MemberAccessExpressionSyntax memberAccess, 
+            out ExpressionSyntax stepValueExpressionOpt,
+            CancellationToken cancellationToken)
+        {
+            // Look for very specific forms.  Basically, only minor variations around:
+            // for (var i = 0; i < expr.Lenth; i++)
+
+            if (forStatement.Declaration != null &&
+                forStatement.Condition.IsKind(SyntaxKind.LessThanExpression) &&
+                forStatement.Incrementors.Count == 1)
+            {
+                var declaration = forStatement.Declaration;
+                if (declaration.Variables.Count == 1)
+                {
+                    var declarator = declaration.Variables[0];
+                    if (declarator.Initializer != null)
+                    {
+                        iterationVariable = declarator.Identifier;
+                        initializer = declarator.Initializer.Value;
+
+                        var binaryExpression = (BinaryExpressionSyntax)forStatement.Condition;
+
+                        // Look for:  i < expr.Length
+                        if (binaryExpression.Left is IdentifierNameSyntax identifierName &&
+                            identifierName.Identifier.ValueText == iterationVariable.ValueText &&
+                            binaryExpression.Right is MemberAccessExpressionSyntax)
+                        {
+                            memberAccess = (MemberAccessExpressionSyntax)binaryExpression.Right;
+
+                            var incrementor = forStatement.Incrementors[0];
+                            return TryGetStepValue(
+                                iterationVariable, incrementor, out stepValueExpressionOpt, cancellationToken);
+                        }
+                    }
+                }
+            }
+
+            iterationVariable = default;
+            memberAccess = default;
+            initializer = default;
+            stepValueExpressionOpt = default;
+            return false;
+        }
+
+        private static bool TryGetStepValue(
+            SyntaxToken iterationVariable, ExpressionSyntax incrementor,
+            out ExpressionSyntax stepValue, CancellationToken cancellationToken)
+        {
+            // support
+            //  x++
+            //  ++x
+            //  x += constant_1
+
+            ExpressionSyntax operand;
+            switch (incrementor.Kind())
+            {
+            case SyntaxKind.PostIncrementExpression:
```

Comment:
Nit: is this indented properly, or will it shift when invoking FormatDocument?

Reply:
It's indented properly.  this is fallout from the IDE actually changing the default for switches at one point.  So it depends on whatever saved preferences have migrated when you were using VS.  Mine is set to "do not indent case" (which is what i actually like as well).

Reply:
these seems also shared using IOperation?

Reply:
VB has no post-increment/decrement operation.  And how they represent step values is very different in the ioperatoin world.  I tried this originally as ioperation and it was more annoying due to the languages actually being so different for this construct.  I like IOperation when it unifies (see the 'PopulateSwitch' refactoring i wrote).  But i don't want to use it if it's not going to actually unify over the languages, and i end up having to write language specific code anyways. :)

Reply:
I see. sure more advanced one is quite different, but I expected the basic form we are dealing with to be quite same. I guess not then.

Reply:
@mavasani @AlekseyTs @333fred @jinujoseph 

feedbacks on loop on IOperation? I remember we talked about in design meeting whether we should make these IOperation more similar each other or separate. looks like our decision to make it separate has some issue expectation people has when they use IOperation.